### PR TITLE
Mushroom people, revived!

### PR DESCRIPTION
**Pull request by ghost**
# KOR SAID IT WAS OK

ONLY FOR PLANETSTATION! You can't get these through magic mirrors, xenobiology, or anything else without admin intervention. I've also blissfully unincluded the spawn for the game panel to avoid admins making a mess out of this race.

remake of bman's pr but with less conflicts hopefully, i'm ready to fight them if they come back.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/datum/species/mush //mush mush codecuck
+	name = "Mushroomperson"
+	id = "mush"
+	mutant_bodyparts = list("caps", "tail_human", "ears", "wings")
+	default_features = list("caps" = "Round", "tail_human" = "None", "ears" = "None", "wings" = "None")
```
Comment (Armhulen) [inline code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: caps won't show up if anyone has an idea on why that is please tell me!

**Comment by ExcessiveUseOfCobblestone** [comment]
> he thinks he can stop the admins

Oh you

**Comment by Armhulen** [comment]
just the dumb ones, and the dumb ones are usually the ones who will spam this 50 rounds in a row before planetstation

**Comment by Repukan** [comment]
Sporecreep

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: why do they have access to ears and tails?

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	hair_color = "FF4B19" //cap color, spot color uses eye color
+	nojumpsuit = 1
+
+	say_mod = "poofs" //what does a mushroom sound like
+	species_traits = list(MUTCOLORS, NOEYES, NO_UNDERWEAR, NOBREATH)
+	speedmod = 1.5 //faster than golems but not by much
+
+	punchdamagelow = 6
+	punchdamagehigh = 14
+	punchstunthreshold = 14 //about 44% chance to stun
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: thats pretty fuckin strong dude

Comment (Armhulen) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: it's what bman had, i could have the species punch way slower but have stronger fists if that would help

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm (hunk 1 above)
Comment (Armhulen) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: lol uuh

**Comment by 81Denton** [comment]
>an alien race that's not humans with animal ears/tail stapled on
>planetstation only

The suffering!

**Comment by Armhulen** [comment]
🙁

**Comment by arsserpentarium** [comment]
Finish snakes!

**Comment by Buggy123** [comment]
sss

File: code/__HELPERS/mobs.dm [hunk 3]
```diff
@@ -72,7 +72,7 @@
 		init_sprite_accessory_subtypes(/datum/sprite_accessory/moth_wings, GLOB.moth_wings_list)
 
 	//For now we will always return none for tail_human and ears.
-	return(list("mcolor" = pick("FFFFFF","7F7F7F", "7FFF7F", "7F7FFF", "FF7F7F", "7FFFFF", "FF7FFF", "FFFF7F"), "tail_lizard" = pick(GLOB.tails_list_lizard), "tail_human" = "None", "wings" = "None", "snout" = pick(GLOB.snouts_list), "horns" = pick(GLOB.horns_list), "ears" = "None", "frills" = pick(GLOB.frills_list), "spines" = pick(GLOB.spines_list), "body_markings" = pick(GLOB.body_markings_list), "legs" = "Normal Legs", "moth_wings" = pick(GLOB.moth_wings_list)))
+	return(list("mcolor" = pick("FFFFFF","7F7F7F", "7FFF7F", "7F7FFF", "FF7F7F", "7FFFFF", "FF7FFF", "FFFF7F"), "tail_lizard" = pick(GLOB.tails_list_lizard), "tail_human" = "None", "wings" = "None", "snout" = pick(GLOB.snouts_list), "horns" = pick(GLOB.horns_list), "ears" = "None", "frills" = pick(GLOB.frills_list), "spines" = pick(GLOB.spines_list), "body_markings" = pick(GLOB.body_markings_list), "legs" = "Normal Legs", "caps" = "Round", "moth_wings" = pick(GLOB.moth_wings_list)))
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/mobs.dm]: pick(GLOB.caps_list) (which will always default to the round until someone adds more)

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: Is this a dark souls reference?

**Comment by ExcessiveUseOfCobblestone** [comment]
The NOBREATHE is a copout/holder until we establish PS atmos correct?

**Comment by Armhulen** [comment]
philip - Today at 3:56 PM
what is the atmos of planetstation, is it just normal air or
Kor - Today at 4:03 PM
@philip just give them nobreath
it will be toxic
philip - Today at 4:03 PM
that's no fun though!
i think them having to use a mask while on the station is a cool interaction
but yeah that for now i guess

**Comment by ExcessiveUseOfCobblestone** [comment]
ko(r)pout

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm (hunk 2 above)
Comment (Armhulen) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: actually genuinely yes

Comment (Armhulen) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: i'll just make a really quick martial art that does mjolnir punches but every punch they make requires a time waiting and that will satisfy balance and my eternal love for the shroom dudes in dark souls 1

File: code/__HELPERS/mobs.dm (hunk 3 above)
Comment (Armhulen) [reply code/__HELPERS/mobs.dm]: that's ok it's inclusive for more caps

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+/datum/species/mush/on_species_loss(mob/living/carbon/C)
+	. = ..()
+	C.faction -= "mushroom"
+
+/datum/species/mush/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+	if(chem.id == "weedkiller")
+		H.adjustToxLoss(3)
+		H.reagents.remove_reagent(chem.id, REAGENTS_METABOLISM)
+		return 1
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: `TRUE`

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm (hunk 2 above)
Comment (Fox-McCloud) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: In my opinion, this is way too strong.

Comment (Armhulen) [reply code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: they get up fast enough to make it only lethal if the victim is right up against a wall, which i don't find to be too problematic

**Comment by Armhulen** [comment]
80% works now i need to fix mush punch

**Comment by Buggy123** [comment]
so... is PS actually going to be a thing? I was under the impression that it was sentenced to eternal development hell.

**Comment by Time-Green** [comment]
Please call them mycaliens instead

File: code/__HELPERS/mobs.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/__HELPERS/mobs.dm]: what did he mean by this...

**Comment by Armhulen** [comment]
@Buggy123 >:(

@Time-Green i mean why not but also why

**Comment by Time-Green** [comment]
@Armhulen because it’s a pun

**Comment by Militaires** [comment]
good job, the hardest part for me to pull off was the mutant parts caps_list bullshit and after i had done it i had burnt out

**Comment by Militaires** [comment]
@Armhulen i suggest you get the basic species merged before moving on

**Comment by Armhulen** [comment]
Good point

File: code/modules/language/mushroom.dm [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/datum/language/mushroom
+	name = "Mushroom"
+	desc = "A language that consists of the sound of periodic gusts of spore-filled air being released."
+	speech_verb = "puffs"
+	ask_verb = "puffs inquisitively"
+	exclaim_verb = "poofs loudly"
+	whisper_verb = "puffs quietly"
+	key = "y"
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/language/mushroom.dm]: 🚲 🏠 but it should be m(ushroom) or p(ff)

File: code/modules/mob/living/carbon/human/species_types/mushpeople.dm [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/datum/species/mush //mush mush codecuck
+	name = "Mushroomperson"
+	id = "mush"
+	mutant_bodyparts = list("caps")
+	default_features = list("caps" = "Round")
+
+	fixed_mut_color = "DBBF92"
+	hair_color = "FF4B19" //cap color, spot color uses eye color
+	nojumpsuit = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/mushpeople.dm]: TRUE

File: code/datums/martial/mushpunch.dm [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+						  "<span class='userdanger'>[A] [atk_verb] you with inhuman strength, sending you flying backwards!</span>")
+		D.apply_damage(rand(15,30), BRUTE)
+		playsound(get_turf(D), 'sound/effects/meteorimpact.ogg', 25, 1, -1)
+		var/throwtarget = get_edge_target_turf(A, get_dir(A, get_step_away(D, A)))
+		D.throw_at(throwtarget, 4, 2, A)//So stuff gets tossed around at the same time.
+		D.Knockdown(20)
+		if(atk_verb)
+			add_logs(A, D, "[atk_verb] (Mushroom Punch)")
+		return 1
+	return 0
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/martial/mushpunch.dm]: FALSE

File: code/datums/martial/mushpunch.dm [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+		D.visible_message("<span class='danger'>[A] [atk_verb] [D] with inhuman strength, sending [D.p_them()] flying backwards!</span>", \
+						  "<span class='userdanger'>[A] [atk_verb] you with inhuman strength, sending you flying backwards!</span>")
+		D.apply_damage(rand(15,30), BRUTE)
+		playsound(get_turf(D), 'sound/effects/meteorimpact.ogg', 25, 1, -1)
+		var/throwtarget = get_edge_target_turf(A, get_dir(A, get_step_away(D, A)))
+		D.throw_at(throwtarget, 4, 2, A)//So stuff gets tossed around at the same time.
+		D.Knockdown(20)
+		if(atk_verb)
+			add_logs(A, D, "[atk_verb] (Mushroom Punch)")
+		return 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/martial/mushpunch.dm]: TRUE

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
ALSO DO THE pick(GLOB.caps_list) >:(

File: code/modules/language/mushroom.dm (hunk 5 above)
Comment (Armhulen) [reply code/modules/language/mushroom.dm]: >:(

File: code/__HELPERS/mobs.dm (hunk 3 above)
Comment (Armhulen) [reply code/__HELPERS/mobs.dm]: what did i mean by this 13 days ago okay i'll add it

**Comment by Armhulen** [comment]
@KorPhaeron They can now never be roundstart races for any reason, ready 4 merge

File: code/modules/language/mushroom.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/language/mushroom.dm]: y :^)

**Comment by ExcessiveUseOfCobblestone** [comment]
c-c-c-can you post pictures

**Comment by Armhulen** [comment]
🙄 fiiine 1 sec

**Comment by Armhulen** [comment]
https://camo.githubusercontent.com/c5f0cd251a5ca6478e7c6cf4fbc6e89b96ba401b/687474703a2f2f692e696d6775722e636f6d2f773336304558742e706e67

this minus spores

**Comment by Armhulen** [comment]
spots??

**Comment by optimumtact** [comment]
@KorPhaeron can you please deal with this

**Pull request merged**